libpod/runtime_ctr.go

```diff
@@ -395,6 +408,14 @@ func (r *Runtime) removeContainer(ctx context.Context, c *Container, force bool)
 		}
 	}
 
+	for _, v := range volumes {
+		if volume, err := runtime.state.Volume(v); err == nil {
```

Comment:
I think this is still not safe. We need some sort of flag to indicate a volume was created with this container and should be destroyed.

Reply:
If the user specifies to remove the volumes associated with the container, he has indicated to the tool to remove the volumes.

Reply:
if err := runtime.removeVolume(ctx, volume, false, true); err != nil && err != ErrNoSuchVolume && err != ErrVolumeBeingUsed {


We now ignore volume does not exist and volume being used

So this will only remove volumes used by the current container being removed that are not used by other containers.